dev/storage.h

```diff
@@ -142,33 +165,79 @@ namespace sqlite_orm {
                 perform_void_exec(db, ss.str());
             }
 #endif
+            void add_generated_cols(std::vector<const table_xinfo*>& columnsToAdd,
+                                    const std::vector<table_xinfo>& storageTableInfo) {
+                //  iterate through storage columns
+                for(size_t storageColumnInfoIndex = 0; storageColumnInfoIndex < storageTableInfo.size();
+                    ++storageColumnInfoIndex) {
+
+                    //  get storage's column info
+                    auto& storageColumnInfo = storageTableInfo[storageColumnInfoIndex];
+                    auto& columnName = storageColumnInfo.name;
+                    if(storageColumnInfo.hidden) {
+                        columnsToAdd.push_back(&storageColumnInfo);
+                    }
+                }
+            }
+            template<class I>
+            void drop_create_with_loss(const I& tImpl, sqlite3* db) {
+                bool started_local_transaction = false;
+                try {
+                    started_local_transaction = this->start_migration();
+
+                    this->drop_table_internal(tImpl.table.name, db);
+                    this->create_table(db, tImpl.table.name, tImpl);
+
+                    this->commit_migration(started_local_transaction);
+                } catch(const std::exception&) {
+                    this->abort_migration(started_local_transaction);
```

Comment:
`abort_migration` will call `ROLLBACK`. Controlling transactions from inside a storage is not a good practice

Reply:
Did you not see I was checking for an existing transaction in start_migration? if there is a transaction in place I don't begin another!